indexer.ts

```diff
@@ -0,0 +1,743 @@
+    let reg = {
+      registrant_id: data._project_id || signerId,
+      status: registry.default_registration_status,
+      submitted_at: new Date(
+        Number(BigInt(block.header().timestampNanosec) / BigInt(1000000))
+      ),
+    };
+    if (data._project_id) {
+      await context.db.Account.upsert({ id: data._project_id }, ["id"], []);
+    }
+
+    await context.db.Account.upsert({ id: signerId }, ["id"], []);
+
+    await context.db.ListRegistration.insert(reg);
+    let activity = {
+      signer_id: signerId,
+      receiver_id: receiverId,
+      timestamp: reg.submitted_at,
+      type: "Register",
+      action_result: reg.registrant_id,
+      tx_hash: receiptId,
+    };
+
+    await context.db.Activity.insert(activity);
+  }
+
+  async function handleProjectRegistrationUpdate(
+    args,
+    receiverId,
+    signerId,
+    receiptId
+  ) {
+    let data = JSON.parse(args);
+    console.log("new Project data::", { ...data }, receiverId);
+    let regUpdate = {
+      status: data.status,
+      admin_notes: data.review_notes,
+      updated_at: new Date(
+        Number(BigInt(block.header().timestampNanosec) / BigInt(1000000))
+      ),
+    };
+
+    await context.db.ListRegistration.update(
+      { registrant_id: data.project_id },
+      regUpdate
+    );
+  }
+
+  async function handlePotApplication(args, receiverId, signerId, receiptId) {
+    let data = JSON.parse(args);
+    console.log("new factory data::", { ...data }, receiverId);
+    await context.db.Account.upsert({ id: data.project_id }, ["id"], []);
+    let application = {
+      pot_id: receiverId,
+      applicant_id: data.project_id,
+      message: data.message,
+      submitted_at: data.submitted_at,
+      current_status: data.status,
+      tx_hash: receiptId,
+    };
+    const appl = await context.db.PotApplication.insert(application);
+
+    let activity = {
+      signer_id: signerId,
+      receiver_id: receiverId,
+      timestamp: application.submitted_at,
+      type: "Submit_Application",
+      action_result: appl[0].id.toString(), // result points to the pot application created.
+      tx_hash: receiptId, // should we have receipt on both action and activity?
+    };
+
+    await context.db.Activity.insert(activity);
+  }
+
+  async function handleApplicationStatusChange(
+    args,
+    receiverId,
+    signerId,
+    receiptId
+  ) {
+    let data = JSON.parse(args);
+    console.log("new factory data::", { ...data }, receiverId);
+
+    let receipt = block
+      .receipts()
+      .filter((receipt) => receipt.receiptId == receiptId)[0];
+    let update_data = JSON.parse(atob(receipt.status["SuccessValue"]));
+    let appl = (
+      await context.db.PotApplication.select({ applicant_id: data.project_id })
+    )[0];
+
+    let applicationReview = {
+      application_id: appl.id,
+      reviewer_id: signerId,
+      notes: update_data.notes,
+      status: update_data.status,
+      reviewed_at: update_data.updated_at,
```

Comment:
`updated_at` won't be present on `update_data`

Reply:
ermm... it won't? the `chef_set_application_status` function returns an application object with those values... including updated at

Reply:
it appears that you are using the function call args, not the result

Reply:
i was using the result, `let update_data = JSON.parse(atob(receipt.status["SuccessValue"]));`

Reply:
oooooops sorry missed that